tests/lit-tests/loop_unroll_varying.ispc

```diff
@@ -0,0 +1,213 @@
+// Test to check '#pragma unroll' functionality for different loop statements
+//   with varying type induction variables.
+
+// RUN: %{ispc} %s --target=host --nowrap -O0 --emit-llvm-text --no-discard-value-names --nostdlib -o - | FileCheck %s -check-prefixes=CHECK_ALL,CHECKO0
+// RUN: %{ispc} %s --target=host --nowrap -O2 --emit-llvm-text --no-discard-value-names --nostdlib -o - | FileCheck %s -check-prefixes=CHECK_ALL,CHECKO2
+
+// XFAIL: LLVM_15_0+
+
+
+#define NUM 1024
+
+
+// CHECK_ALL-LABEL: define void @foo_for___
+
+// CHECKO0:         for_loop:
+// CHECKO0:         call void @goo_for___
+// CHECKO0-NOT:     call void @goo_for___
+// CHECKO0:         br label %{{[a-zA-Z_][a-zA-Z0-9_]*}}
+// CHECKO0:         }
+
+// CHECKO2:         for_loop{{.*}}:
```

Comment:
Is this an attempt to match basic block label?

Isn't it enough to just count the number of function calls in the function?

Reply:
Its not sufficient because the compiler generates different basic blocks depending on the mask or lane status. In those cases, counting may cause false passing tests because some instructions are repeated across basic blocks (not due to unrolling). In some edge cases, optimizations can cause entire sections of these duplicated basic blocks not to be present, making counting even trickier. There may be a better way to use the FileCheck syntax to accomplish this though.

Reply:
What about defining the function `unmasked`? Would it help?

Reply:
For `for` varying loops, the unrolling is generating a basic block for each unrolled section (i.e. a branch separates each unrolled instruction from it's unrolled copies). This may have negative impacts on the performance benefits to be gained by the transformation.

For `foreach` loops, the unrolled instructions copies are generally (there are cases that break this, e.g. [odd number unroll factors](https://github.com/e3m3/ispc/blob/6a441e9b5fa3f2c344b4a1092e6d28639ce3b4ce/tests/lit-tests/loop_unroll_varying.ispc#L33)) in the same basic block.

Reply:
> What about defining the function `unmasked`? Would it help?

I'll try this.

Reply:
> What about defining the function `unmasked`? Would it help?

Neither of these versions helped.

```cpp
extern void goo_for(int);
void foo_for() {
    unmasked {
        #pragma unroll(3)
        for (int iter1 = 0; iter1 < NUM; iter1++) {
            goo_for(iter1);
        }
    }
}
```
```cpp
extern void goo_for(int);
void foo_for() {
    #pragma unroll(3)
    for (int iter1 = 0; iter1 < NUM; iter1++) {
        unmasked {
            goo_for(iter1);
        }
    }
}
```